Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -1215,38 +1245,16 @@ void translateRange(const std::string &instructions,
     outGroups.push_back(newGroup);
   }
 }
-} // anonymous namespace
-
-/**
- * Used to validate the inputs for GroupDetectors2
- *
- * @returns : A map of the invalid property names to what the problem is.
- */
-std::map<std::string, std::string> GroupDetectors2::validateInputs() {
-  std::map<std::string, std::string> errors;
-
-  const std::string pattern = getPropertyValue("GroupingPattern");
-
-  boost::regex re(
-      "^\\s*[0-9]+\\s*$|^(\\s*,*[0-9]+(\\s*(,|:|\\+|\\-)\\s*)*[0-9]*)*$");
-  if (!pattern.empty() && !boost::regex_match(pattern, re)) {
-    errors["GroupingPattern"] =
-        "GroupingPattern is not well formed: " + pattern;
-  }
-
-  return errors;
-}
 
 /**
- * Translate the PerformIndexOperations processing instructions into a format
- * usable by GroupDetectors.
- *
- * @param instructions : Instructions to translate
- * @param commands : A stringstream to be filled
- */
-void GroupDetectors2::translateInstructions(const std::string &instructions,
-                                            std::stringstream &commands) {
-  // vector of groups, each group being a vector of its spectra
+* Translate the PerformIndexOperations processing instructions into a vector
+*
+* @param instructions : Instructions to translate
+* @return : A vector of groups, each group being a vector of its 0-based
+* spectrum indices
+*/
+std::vector<std::vector<int>>
+translateInstructions(const std::string &instructions) {
```

Comment:
Why not move this anonymous namespace these functions are defined in to the top of the file so two separate anonymous workspaces aren't required?